sky/task.py

```diff
@@ -624,13 +625,34 @@ def service(self) -> Optional[service_spec.SkyServiceSpec]:
     def set_service(self,
                     service: Optional[service_spec.SkyServiceSpec]) -> 'Task':
         """Sets the service spec for this task.
-
         Args:
           service: a SkyServiceSpec object.
 
         Returns:
           self: The current task, with service set.
         """
+
+        if (service and service.spot_placer is not None):
+
+            launchable_resources, _, _ = (
+                optimizer.fill_in_launchable_resources(
```

Comment:
could we use `cloud.get_feasible_launchable_resources` instead? Reason: avoid import optimizer in task

Reply:
Sounds good.